calabash/Classes/UISpec/UIScriptASTWith.m

```diff
@@ -77,6 +78,55 @@ - (NSString *) description {
   }
 }
 
+- (NSArray *)handleIFrameQueryFromIFrameResult:(NSDictionary *)iFrameResult {
+  NSDictionary *iframeInfo = iFrameResult[IFRAME_INFO_KEY];
+  
+  LPWebQueryType queryType = [iframeInfo[QUERY_TYPE_KEY] unsignedIntegerValue];
+  UIView<LPWebViewProtocol> *webView = iFrameResult[WEBVIEW_KEY];
+  NSString *iframeSelector = iframeInfo[QUERY_KEY];
+  
+  if (iframeSelector == nil) {
+    LPLogError(@"Missing query string for IFrame Query");
+    return [NSArray array];
+  } else if (webView == nil) {
+    LPLogError(@"Missing webview for IFrame query");
+    return [NSArray array];
+  }
+  
+  NSArray *results = [LPWebQuery arrayByEvaluatingQuery:(NSString *)self.objectValue
+                                          frameSelector:iframeSelector
+                                                   type:queryType
+                                                webView:webView
+                                       includeInvisible:YES];
+  /*
+   *  And now we must offset the iframe query results by the frame (left, top) of the original 
+   *  iframe.
+   */
+  NSMutableArray *ret = [NSMutableArray arrayWithCapacity:results.count];
+  
+  NSDictionary *iFrameRect = iFrameResult[@"rect"];
+  float xOffset, yOffset;
+  
+  xOffset = [iFrameRect[@"left"] floatValue];
+  yOffset = [iFrameRect[@"top"] floatValue];
+  
+  for (NSDictionary *result in results) {
+    NSMutableDictionary *mResult = [result mutableCopy];
+    
+    NSMutableDictionary *mRect = [mResult[@"rect"] mutableCopy];
+    mRect[@"x"] = @([mRect[@"x"] floatValue] + xOffset);
+    mRect[@"left"] = @([mRect[@"left"] floatValue] + xOffset);
+    mRect[@"center_x"] = @([mRect[@"center_x"] floatValue] + xOffset);
+    mRect[@"y"] = @([mRect[@"y"] floatValue] + yOffset);
+    mRect[@"top"] = @([mRect[@"top"] floatValue] + yOffset);
+    mRect[@"center_y"] = @([mRect[@"center_y"] floatValue] + yOffset);
+    
```

Comment:
@jmoody Can you sanity check my math here? Will this cause any issues with scaling on different devices?